pan-domain-auth-core/src/main/scala/com/gu/pandomainauth/model/AuthenticatedUser.scala

```diff
@@ -5,4 +5,5 @@ import java.util.Date
 case class AuthenticatedUser(user: User, authenticatingSystem: String, authenticatedIn: Set[String], expires: Long, multiFactor: Boolean) {
 
   def isExpired = expires < new Date().getTime
+  def hasExpiryExtension(extension: Long) = (expires + extension) > new Date().getTime
```

Comment:
I'd second using grace throughout, extension make it sound like you're bumping the expiry date but grace / gracePeriod reflect what's going on.

isWithinExpiryGracePeriod could work as a method name.